optimum/onnxruntime/modeling_decoder.py

```diff
@@ -623,7 +623,7 @@ def _from_transformers(
         task: Optional[str] = None,
     ) -> "ORTModelDecoder":
         if task is None:
-            task = cls._AUTOMODELS_TO_TASKS[cls.auto_model_class]
+            task = cls._AUTOMODELS_TO_TASKS[cls.auto_model_class.__name__]
```

Comment:
Maybe we can make a method that would do that, and avoid this, which is "too aware" of the underlying implementation IMO.

Something like: 
```
cls._auto_models_to_task(auto_model_class)
```